### PR TITLE
[capture] Fixes AES fixed-vs-random non-batch capture

### PR DESCRIPTION
**Pull request by vrozic**
At the moment AES fixed-vs-random non-batch capture doesn't work properly.
The problem is that the cipertext variable is updated too soon, so the received ciphertext is compared with the result of the following encryption, causing `Uncorrect encryption result` error.

This PR fixes it.
The patch is not the most elegant, but things will simplify later on when I modify `data_generator.py` to include random capture.

Thanks @m-temp for pointing out this problem

### CODE REVIEW

**Review by nasahlpa — APPROVED** [review]
LGTM, thanks! Yes, I agree, we should switch as soon as possible to the data generator class.

**Review by m-temp — APPROVED** [review]
Thx @vrozic !
Would it generally make sense to swap "gen_ref_crypto" with "start_capture"?
This would mean, we always generate the current values and not the next ones.
The drawback would be, that gen_reg_crypto cannot run in parallel to the processing on the device. Don't know how bad this would be, but it would generally easy the understanding of the program flow.

Generally, I find it hard to understand why (if) the program flow is correct in terms of the waves/pt/ct/key quadruple. Perhaps we could add some explaining comments, what is happening where and why. I guess one problem is that the different behaviors are not aligned with either batch vs. non-batch or fsvr v.s random, but there are 3 different flows for 4 tests.

**Comment by vrozic** [comment]
> Would it generally make sense to swap "gen_ref_crypto" with "start_capture"?

Yes, this absolutely makes sense.
At the moment it is tricky to do due to a strange way in which we generate data for aes_random capture (with old ciphertext = new plaintext).
I would rather first modify random capture to also use data_generator. This should simplify the code and reduce the number of flows.

File: capture/capture_aes.py [hunk 1]
```diff
@@ -282,12 +288,13 @@ def capture(scope: Scope, ot_aes: OTAES, capture_cfg: CaptureConfig,
 
             # Generate reference crypto material and store trace.
             for i in range(capture_cfg.num_segments):
-                text, key, ciphertext, sample_fixed = generate_ref_crypto(
-                    sample_fixed = sample_fixed,
-                    mode = capture_cfg.capture_mode,
-                    key = key,
-                    plaintext = text
-                )
+                if capture_cfg.batch_mode or capture_cfg.capture_mode == "aes_random":
```
[the remark this answers is not in the document]
Comment (vrozic) [reply capture/capture_aes.py:294]: Yes. I've added a comment now

**Pull request merged**